### DISCUSSION TITLE
Phalcon v5.0.2 + PHP 8.1 - Model caching error

### DISCUSSION THREAD

**Discussion by Alistar84** [phalcon/cphalcon#16130] [Q&A]
Hello everybody,
I have a problem with model caching.

In v4 all my models extend a class that implements a force cache system similar to this one [ https://docs.phalcon.io/5.0/en/db-models-cache#forcing-cache](url)

Now, in v5, if I run a find or findFirst that is already cached, I've got this error:
`Fatal error: Uncaught TypeError: unserialize(): Argument #1 ($data) must be of type string, array given.`

How can I solve?

**Comment by niden** [comment]
@Alistar84 Moving this to an issue.
